### PR TITLE
Functionality to create Paste

### PR DESCRIPTION
**Pull request by maiyuki**

### CODE REVIEW

File: phabfive/paste.py [hunk 1]
```diff
@@ -32,6 +35,48 @@ def _convert_ids(self, ids):
...
+
+        with open(file, "r") as f:
+            text = f.read()
+
+        transactions = []
+        transactions_values = [
+            {"type": "title", "value": title},
+            {"type": "text", "value": text},
+            {"type": "language", "value": language},
+            # TODO: Create a function that vaildates tags' and 'subscribers' existence
```
Comment (holmboe) [inline phabfive/paste.py]: Speling

File: phabfive/paste.py [hunk 2]
```diff
@@ -32,6 +35,48 @@ def _convert_ids(self, ids):
...
+        ]
+        # Phabricator does not take None (empty list is ok for projects/subscribers) as a value, therefor only "type" that has valid value can be sent as an argument
+        for item in transactions_values:
+            if None not in item.values():
+                transactions.append(item)
+        try:
+            id_and_phid = self.phab.paste.edit(transactions=transactions)
+        except APIError:
+            raise PhabfiveDataException(
+                "No valid input, maybe unvalid tags/subscribers or other error"
```
Comment (holmboe) [inline phabfive/paste.py]: "Invalid input"

Is it possible to gather information from the `APIError` what kind of problem caused this? If so, then that should probably be passed through to `PhabfiveDataException`.

File: phabfive/cli.py [hunk 3]
```diff
@@ -78,13 +78,20 @@
...
     <ids> ...            Paste monogram (P123), example P1 P2 P3
+    <title>              Title for Paste
+    <file>               A file with text content for Paste ex. myfile.txt
 
 Options:
     -h, --help           Show this help message and exit
+
+Paste Create Options:
+    -t, --tags=<args> ...           Project name(s), ex. --tags=projectX,projectY,projectZ
+    -s, --subscribers=<args> ...    Subscribers - user, project, mailing list name. Ex --subscribers=user1,user2,user3
```
Comment (holmboe) [inline phabfive/cli.py]: `<args>` is a very programmer-esque term. Could you use `tagX,projectY` or similar?

File: phabfive/cli.py [hunk 4]
```diff
@@ -78,13 +78,20 @@
 sub_paste_args = """
 Usage:
     phabfive paste list
+    phabfive paste create <title> input <file> [options]
```
Comment (holmboe) [inline phabfive/cli.py]: I vore for removing `input` all together here, and just go with `phabfive paste create <title> <file>`. At least I can't see why not right now.

**Comment by holmboe** [comment]
Also, update README?

File: phabfive/paste.py (hunk 2 above)
Comment (maiyuki) [reply phabfive/paste.py]: I will then change to 
```
except APIError as a:
    raise PhabfiveDataException(a)
```
which results in:
```
~/dynamist/phabfive(feature/create-paste*) » phabfive paste create new file.txt --subscribers=testuser1235790
ERR-CONDUIT-CORE: Edges are not available for objects of type '????'!

~/dynamist/phabfive(feature/create-paste*) » phabfive paste create new file.txt --tags=qjk
ERR-CONDUIT-CORE: Edge transactions must have destination PHIDs as in edge lists (found key "qjk" on transaction of type "41").
```

Comment (maiyuki) [reply phabfive/paste.py]: Do you have suggestion on how to remove `ERR-CONDUIT-CORE:`? :) `a.replace('ERR-CONDUIT-CORE:', '')`  does not apply

Comment (maiyuki) [reply phabfive/paste.py]: `AttributeError: 'APIError' object has no attribute 'lstrip'`
`a` is not a string

Comment (maiyuki) [reply phabfive/paste.py]: Used `replace()` because `lstrip()` weirdly strip more that what it is suppose to do.

File: phabfive/paste.py [hunk 5]
```diff
@@ -32,6 +35,46 @@ def _convert_ids(self, ids):
...
+            {"type": "text", "value": text},
+            {"type": "language", "value": language},
+            # TODO: Create a function that vaildates tags' and 'subscribers' existence
+            {"type": "projects.add", "value": tags},
+            {"type": "subscribers.add", "value": subscribers},
+        ]
+        # Phabricator does not take None (empty list is ok for projects/subscribers) as a value, therefor only "type" that has valid value can be sent as an argument
+        for item in transactions_values:
+            if None not in item.values():
+                transactions.append(item)
```
Comment (Grokzen) [inline phabfive/paste.py]: Can be simplified to list comprehension `transactions = [item for item in transaction_values if None not in item.values()]`

**Pull request merged**